### DISCUSSION TITLE
Dynamic AR and OD mod depending on current strain difficulty

### DISCUSSION THREAD

**Discussion by finesseandstyle** [ppy/osu#19920] [Ideas] [answered]
![image](https://user-images.githubusercontent.com/41844929/186054878-f5f3d679-121a-41c3-a9a7-2a8c1442ef68.png)

I want to propose a mod that dynamically adjusts current AR and OD in real time based if you're currently in a difficulty spike or in a slow section. Difficulty spikes imply more dense and/or spaced objects so the purpose of this mod is to have a consistent note density throughout the whole map.

Slow sections are also less pp rewarding meaning having a lower OD wouldn't impact the performance if it was as high as the hardest sections of the map.

Very high No Mod SR maps (>9*) could be more playable with this mod where the AR is capped at 10 (e.g. where it could jump to 10.5 at diff spikes) and the note density is high.

Please tell me if this is a terrible idea like it would screw up the timing or be completely unplayable.

**Comment by iSimpFor** [comment]
maybe add a min AR(for the least dense sections) and max AR(for the most dense) mod setting for people to scale dynamic AR to their preferences

**Reply by nagi-desuuu** [reply]
This would be okay if it were it's own mod.

**Reply by iSimpFor** [reply]
i think making it its own mod would be redundant since the ar values would have to be set somewhere and people would like to change them to their own values, regardless of the initial values

**Comment by smoogipoo** [comment, marked as answer]
This is not going to happen due to technical complexity.
